### PR TITLE
[Enhancement] Support expression conversion of struct sub-column in Iceberg 

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 1]
```diff
@@ -407,5 +419,17 @@ public String visitVariableReference(ColumnRefOperator operator, Void context) {
         public String visitCastOperator(CastOperator operator, Void context) {
             return operator.getChild(0).accept(this, context);
         }
+
+        public String visitSubfield(SubfieldOperator operator, Void context) {
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:423]: return qualified name

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java [hunk 2]
```diff
@@ -122,7 +124,16 @@ private static class IcebergExprVisitor extends ScalarOperatorVisitor<Expression
 
         private static Type.TypeID getResultTypeID(String columnName, IcebergContext context) {
             Preconditions.checkNotNull(context);
-            return context.getSchema().fieldType(columnName).typeId();
+            return getColumnType(columnName, context).typeId();
+        }
+
+        private static Type getColumnType(String qualifiedName, IcebergContext context) {
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java:130]: this is compatible with previous behavior

**Review by Smith-Cruise — APPROVED** [review]

**Review by zombee0 — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 19 / 20 (95.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java|19|20|95.00%|[426]|

**Review by dirtysalt — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by letian-jiang** [comment]
@mergify backport branch-3.2

**Comment by letian-jiang** [comment]
@mergify backport branch-3.1

**Comment by letian-jiang** [comment]
backported to 3.1 by PR https://github.com/StarRocks/starrocks/pull/33898

**Pull request merged**